### PR TITLE
Added code coverage, fixed makefile, cleaned the github workflow

### PR DESCRIPTION
**Pull request by kkapka**
## Type of changes

- [ ] Bug fix
- [ ] New feature
- [ ] Documentation / docstrings
- [x] Tests
- [x] Other

## Checklist

- [ ] I've run the latest [black](https://github.com/ambv/black) with default args on new code.
- [x] I've updated CHANGELOG.md and CONTRIBUTORS.md where appropriate.
- [ ] I've added tests for new code.
- [x] I accept that @willmcgugan may be pedantic in the code review.

## Description

**Why?**

I tried to write unit tests for code coverage gaps. Unfortunately, the test target in makefile was not working, also I didn't know how to build an app locally on my laptop. So I decided to fix these problems

**How?**

- I made a `build` target in `Makefile` to allow everyone to build the app easily (I called that target in GH workflow). 
- I fixed a test target by adding `pytest-cov` dependency. 
- I fixed the dependency `sphinx-rtd-theme` for docs generation. 
- I split requirements into 'target' files and I put them into `requirements` directory.
- I made a package from the tests directory to allow counting code coverage for the source code (And fixed imports of local modules in the tests)
- I added `codecov` step for GH workflow and after merge to `master` we can add badge for code coverage in the `README.md` :)

**Result?**

- GH workflows should pass on three python versions
image

- We can see the codecov step and code coverage report
image

**Future?**

- After merge to master we can add `codecov` bage to `README.md`

### CODE REVIEW

**Comment by kkapka** [comment]
link to checks on my fork -> https://github.com/kkapka/rich/pull/2/checks

File: .github/workflows/pythonpackage.yml [hunk 1]
```diff
@@ -16,17 +16,16 @@ jobs:
...
       - name: Test with pytest
         run: |
-          pip install poetry pytest
-          poetry install
-          pip install .
-          pytest tests/ -v
+          make test
+      - uses: codecov/codecov-action@v1
+        with:
+          file: ./cov.xml
```
Comment (willmcgugan) [inline .github/workflows/pythonpackage.yml:30]: Where is cov.xml that is referenced here?

Comment (kkapka) [reply .github/workflows/pythonpackage.yml:30]: Hi, that coverage report is generated here (argument `xml:cov.xml`) -> https://github.com/willmcgugan/rich/pull/63/files#diff-b67911656ef5d18c4ae36cb6741b7965R12

File: tests/test_bar.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 from rich.bar import Bar
 
-from render import render
+from .render import render
```
Comment (willmcgugan) [inline tests/test_bar.py:5]: Curious why does this not fail with "attempted relative import with no known parent package" ?

There's some code at the bottom of some of these files which generates the expected output. I run them with "python test_bar.py", which will break if there are relative imports.

**Comment by willmcgugan** [comment]
You closed? Sorry if I was taking my time, I was planning on merging...

**Pull request closed without merging**